app/views/shared/_sidebar_collection.html.erb

```diff
@@ -1,14 +1,16 @@
 <div class="header">
   <div class="header-inner">
+    <h3><%= local_assigns[:index_path] ? link_to(name, index_path) : name %></h3>
     <div class="options">
       <%= render 'shared/add_item_dropdown',
         name: name,
         new_path: new_path,
-        category_id: category_id,
-        menu_class: 'pull-right'
+        category_id: category_id
       %>
+      <% if name =='Issues' %>
```

Comment:
I don't _love_ that we're checking a string to see if we want to add additional partials. I'm not sure I have a better idea, as much as a different idea.

**issues/_sidebar.html.erb**
```erb
<%= render 'shared/sidebar_collection',
  name: 'Issues',
  collection: @issues,
  new_path: main_app.new_project_issue_path(current_project),
  category_id: nil  do
    render('import_box')
  end
%>
```

**shares/_sidebar_collection.html.erb**
```erb
...
    <div class="options">
      <%= render 'shared/add_item_dropdown',
        name: name,
        new_path: new_path,
        category_id: category_id
      %>
      <%= yield %>
    </div>
...
```

If no block is passed the yield is simply ignored. If a block is passed it'll render the given block. In this case rendering an additional partial.
I like it because it keeps the knowledge at the caller, and the sidebar doesn't need to know about what it is rendering.
It could also be a little off putting to come across in the code, so I think a second opinion would be good.

Reply:
When I implement these 2 code blocks in the respective files, the down-caret doesn't show up and the import box doesn't render. Maybe I missed something?

Reply:
We clarified the issue was the usage of `<% yield %>` where it should be `<%= yield %>`. Got it working now.